### ISSUE TITLE
Feature suggestion: Provide page title as placeholder for "Was this page helpful"

### ISSUE THREAD

**Issue by wilhelmer** [squidfunk/mkdocs-material#3258]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I want to suggest an idea and checked that ...

- [X] ... to my best knowledge, my idea wouldn't break something for other users
- [X] ... the documentation does not mention anything about my idea
- [X] ... there are no open or closed issues that are related to my idea

### Description

When using the "Was this page helpful" feature, it's possible to pre-fill forms with the path to the current page by using the {} placeholder, as shown in the [docs](https://squidfunk.github.io/mkdocs-material/setup/setting-up-site-analytics/?h=page+helpful#was-this-page-helpful).

```
https://github.com/<username>/<repository>/issues/new/?title=[Feedback]+{}
```

As discussed in #3166, I suggest one of the following:

a) Use the page **title** instead of the **path** for the `{}` placeholder
b) Provide two placeholders: `{title}` and `{path}`.

### Use Cases

In the example above, 

> [Feedback] My Topic

is much more readable than 

> [Feedback] /my/project/dir/misc/docs/my-topic.html

**Comment by squidfunk** [comment]
Fixed in [`4ac83e2a6`](https://github.com/squidfunk/mkdocs-material-insiders/commit/4ac83e2a64cc2189822a4f71aa744dbec65dc7bf). I switched to named placeholders which are surprisingly supported in Jinja – pretty cool! This means we can add even more attributes if necessary. The old syntax is still supported, but now deprecated.

#### New syntax

```
https://example.com/?url={url}&title={title}
```

#### Old syntax

```
https://example.com/?url={}
```

Parameters can be defined or omitted, so it's also possible to solely use the `title`. Order doesn't matter. [The documentation has been updated](https://squidfunk.github.io/mkdocs-material/setup/setting-up-site-analytics/#feedback-rating-note) to explain the new syntax with an example.

**Comment by squidfunk** [comment]
The changes have been released as part of 8.0.2+insiders.4.1.0

**Comment by wilhelmer** [comment]
Fantastic, thank you!

**Comment by wilhelmer** [comment]
Unfortunately, it's using the nav title, not the page title. Would be great if you could fix that. Thanks 🙏 

<img width="1276" alt="Bildschirmfoto 2021-12-01 um 09 33 11" src="https://user-images.githubusercontent.com/2234833/144199990-3f7ccef0-7cc2-4091-8f75-2e5e4584ec73.png">

**Comment by squidfunk** [comment]
Hmm, we use `page.title`, which is sourced from `mkdocs.yml` and/or from front matter. Does it work if you set the title via metadata? If you've defined the `h1` in Markdown, we would have to parse Markdown/HTML, which I wouldn't want to do.

**Comment by wilhelmer** [comment]
How about something like this?

```jinja
{% if page.toc|first is defined %}
  {% set _title = page.toc.items[0].title %}
{% else %}
  {% set _title = page.title|striptags %}
{% endif %}
```

**Comment by squidfunk** [comment]
Well, some users omit the page title starting with `h2`, so I don't think it's generalizable.

**Comment by wilhelmer** [comment]
Maybe the precedence should be metadata > h1 > nav title. 

So if users start with `h2`, they can still control the page title via metadata or nav title.

Anyway, not your issue. Would still be nice to hear your thoughts on this.

**Comment by squidfunk** [comment]
I don't think that's possible, as MkDocs will automatically override the `page.title` with the value you set in front matter, so retrieving the original page title from the template isn't likely to be possible. Therefore, interspersing `h1` wouldn't work. I may be wrong thou, need to check first.

**Comment by wilhelmer** [comment]
My point was that the precedence should be changed in MkDocs, not in Material. MkDocs should always set `page.title` to the metadata title, if not exists to the `h1` text, if not exists to the nav title. I'd create an issue in their repo, but was unsure whether my approach made sense.

**Comment by squidfunk** [comment]
Ah okay, that makes more sense, yes. Yes, the precedence makes sense to me.

**Comment by squidfunk** [comment]
Okay, so I've decided to not implement the proposed solution for now. It's a significant deviation from how we handle titles, and if we implement it for the feedback widget, we must also change it in other parts of the theme as well. However, I've added support for setting it via metadata (wasn't working before) in the last commit, so if you set the title via metadata.

We can discuss over at the MkDocs repository regarding the precedence of page titles. This should be implemented in MkDocs, not Material for MkDocs, as it concerns all themes.
